CHANGELOG/v1.10.0-beta.1.md

```diff
@@ -0,0 +1,449 @@
+🚨 This is a BETA RELEASE. Use it only for testing purposes. If you find any bugs, file an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/new).
+
+:warning: **BETA RELEASE NOTES** :warning:
+
+## Highlights
+- Bumped to controller-runtime v0.20, k8s.io/* v0.32, controller-gen v0.17
+- Features:
+  - ClusterResourceSet was promoted to GA (#11365)
+  - PriorityQueue was added as an alpha feature (see kubernetes-sigs/controller-runtime#2374 for more details) (#11698)
+- API: Added additional validation to our CRDs (#11834)
+- API: Various improvements to v1beta2 conditions
+- ClusterClass: Support referencing ClusterClasses across namespaces (#11352 #11395 #11649)
+- ClusterClass: Add NamingStrategy for InfraCluster (#11898)
+- Machine: Add MachineDrainRule behavior "WaitCompleted" (#11545)
+- Machine: Add --additional-sync-machine-labels flag to allow syncing additional labels to Nodes (#11650)
+- Machine: Sync Machine annotations to Nodes (#11813)
+- MachineSet/MachineDeployment: Add NamingStrategy to MachineDeployment (#11172)
+- MachineSet/KCP: Improve preflight checks (#11927 #11941 #11928)
+- CRD migration: Add CRD migrator (#11889)
+- Metrics: Add ClusterCache (#11789) and SSA cache (#11635) metrics
+- Merged CAPIM in CAPD (see From CAPD(docker) to CAPD(dev) for more details)
```

Comment:
Can we keep the link like here: https://github.com/kubernetes-sigs/cluster-api/issues/11656#issuecomment-2743361589? I'm not sure if folks will find the proposal without it